File: src/main/java/com/range/rpms/user/dao/repository/UserRepository.java
package com.range.rpms.user.dao.repository;

import com.range.rpms.user.dao.model.User;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;

public interface UserRepository extends JpaRepository<User,Long> {
    boolean existsByUsername(String username);

    boolean existsByEmail(String email);

    Optional<User> findByUsername(String username);
}

File: src/main/kotlin/com/range/rpms/users/domain/repository/UserRepository.kt
package com.range.rpms.users.domain.repository

import com.range.rpms.users.domain.entity.User
import org.springframework.data.jpa.repository.JpaRepository
import java.util.Optional
import java.util.UUID

interface UserRepository : JpaRepository<User, UUID> {
    fun findByUsername(username: String):User?
    fun findByEmail(email: String): User?
    fun existsByUsername(username: String): Boolean
    fun existsByEmail(email: String): Boolean
}